File: bukkit/src/main/java/io/turbo/clientblockcommon/listeners/PlayerListener.java
package io.turbo.clientblockcommon.listeners;

import io.turbo.clientblockcommon.config.BlockedClients;
import io.turbo.clientblockcommon.config.WhitelistedPlayers;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class PlayerListener implements Listener {

    @EventHandler
    public void on(PlayerJoinEvent event) {
        if (WhitelistedPlayers.i().getWhitelistedPlayers().contains(event.getPlayer().getName())) return;
        BlockedClients.i().getBlockedClients().forEach(clientName -> {
          if (clientName.equals(event.getPlayer().getClientBrandName())) event.getPlayer().kick();
        });
    }
}

File: common-kotlin/src/main/kotlin/io/turbo/clientblockkotlin/listeners/PlayerListener.kt
package io.turbo.clientblockkotlin.listeners;

import io.turbo.clientblockkotlin.config.BlockedClients
import io.turbo.clientblockkotlin.ClientBlock
import io.turbo.clientblockkotlin.config.WhitelistedPlayers
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class PlayerListener : Listener {

    @EventHandler
    fun on(event: PlayerJoinEvent) {
      if (!(event.player.isOnline)) return;
      if (WhitelistedPlayers().WhitelistedPlayers.contains(event.player.name)) return;
        BlockedClients.i()?.blockedClients?.forEach { string ->
            if (event.player.clientBrandName.equals(string)) {
                event.player.kick()
                ClientBlock().logger.info {
                    " Player $event.player.name (IP: $event.player.address) was kicked for logging in on blacklisted client $string"}
        }
      }
    }
}

